### PR TITLE
Gh7 version is integer

### PR DESCRIPTION
**Pull request by 030**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -34,215 +34,228 @@
...
-    default: {
-      fail("Unsupported version: ${version}.  Implement me?")
-    }
-  }
-
-  $java_home = "${install_dir}/jdk1.${version}.0_${javaUpdate}"
-  $javaDownloadURI = "http://download.oracle.com/otn-pub/java/jdk/${version}u${javaUpdate}-b${javaBuild}/jdk-${version}u${javaUpdate}-linux-${arch}.rpm"
-  $jceDownloadURI = 'http://download.oracle.com/otn-pub/java/jce/8/jce_policy-8.zip'
+class jdk_oracle (
+    $version      = 8,
```
Comment (schrepfler) [inline manifests/init.pp]: Wouldn't this mean the version of the JDK would be permanently set to 8?

Comment (schrepfler) [inline manifests/init.pp]: Additionally, wouldn't this mean anyone who has explicitly set the version would have to perform a change as their configuration would no longer be compatible?

**Comment by 030** [comment]
If jdk_oracle::version is set in hiera then it will be overwritten, but indeed if it is set as a string in hiera then it will be incompatible.

**Comment by 030** [comment]
If we would merge this then the it is an API change and then it would be 1.0.0

**Comment by schrepfler** [comment]
Ok, so we can bump the major version and have a notice in the release notes.

**Comment by schrepfler** [comment]
Merged, let's see how to make the break in compatibility apparent.

**Comment by 030** [comment]
Yes. I have done the same in this module https://github.com/030/puppet-sftp/blob/master/manifests/init.pp#L13. If ```sftp::group``` is defined in hiera then  it will be overwritten

**Pull request merged**